cmd/client/main.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"flag"
+	"log"
+)
+
+func main() {
+	address := flag.String("address", "", "The address for the chat room as ip:port")
+	room := flag.String("room", "", "The room number")
+
+	flag.Parse()
+
+	if *address == "" {
+		log.Fatal("Address must be provided!")
+	}
+
+	if *room == "" {
+		log.Fatal("Room number must be provided!")
+	}
+
+	tcpClient := Client{
```

Comment:
Maybe use the interface  you wrote ? what is the type of  tcpClient ?
look here https://tour.golang.org/methods/10